zephyr/include/sof/lib/regions_mm.h

```diff
@@ -10,31 +10,122 @@
 
 #include <adsp_memory.h>
 #include <adsp_memory_regions.h>
+#include <ipc/topology.h>
+#include <rtos/alloc.h>
+#include <sof/list.h>
 #include <zephyr/drivers/mm/system_mm.h>
-#include <zephyr/sys/mem_blocks.h>
 #include <zephyr/init.h>
-#include <ipc/topology.h>
+#include <zephyr/sys/mem_blocks.h>
 
-/*
- * Struct containing information on virtual memory heap.
- * Information about allocated physical blocks is stored in
- * physical_blocks_allocators variable and uses zephyr memblocks api.
+/* Dependency on ipc/topology.h created due to memory capability definitions
+ * that are defined there
+ */
+
+/* API is NOT re-entry safe.
+ * This is due to our assumption that only management code will ever handle
+ * memory operations on heaps themselves.
+ */
+
+/* Defines maximum amount of memory block allocators in one heap
+ * since minimum realistic block should be cache line size
+ * and block sizes in allocator must be powers of 2
+ * so logically it gives limited block sizes
+ * eg allocator of block size 64 128 256 512 2048 4096 1024 8192
+ * would end up in 8 allocators.
+ * It is expected that allocations bigger than that would
+ * either be spanned on specifically configured heap or have
+ * individual configs with bigger block sizes.
+ */
+#define MAX_MEMORY_ALLOCATORS_COUNT 8
+
+/* vmh_get_default_heap_config() function will try to split the region
+ * down the given count. Only applicable when API client did not
+ * use its config.
+ */
+#define DEFAULT_CONFIG_ALOCATORS_COUNT 5
+
+/** @struct virtual_memory_heap
+ *
+ *  @brief This structure holds all information about virtual memory heap
+ *  it aggregates information about its allocations and
+ *  physical mappings.
+ *
+ *  @var node generic list member used for list operations
+ *  @var virtual_region pointer to virtual region information, it holds its
+ *  attributes size and beginning ptr provided by zephyr.
+ *  @var physical_blocks_allocators[] a table of block allocators
+ *  each representing a virtual regions part in blocks of a given size
+ *  governed by sys_mem_blocks API.
+ *  @var allocation_sizes[] a table of bit arrays representing sizes of allocations
+ *  made in physical_blocks_allocators directly related to physical_blocks_allocators
+ *  @var span configuration value deciding if heap allocations
+ *  will be contiguous or single block.
  */
 struct virtual_memory_heap {
```

Comment:
As long as you call it ```vmh_something``` as this make it obvious to developers that this structure is part of the virtual heaps core API.

Reply:
But if i change it here the vmh could as well be any other random string since the vmh would not relate to any full description of anything. Any new person looking at vmh_heap will be asking themselves what the vmh stands for.

Reply:
Pls use ```vmh_``` prefix for all structs, APIs and macros. We need to be more obvious. Easy change.

Reply:
I will do it if You insist but as I stated this code will be less approachable for new people.